app/controllers/admin/assets_controller.rb

```diff
@@ -180,6 +180,41 @@ def refresh_active_encode_status
     redirect_to admin_asset_url(status.asset), notice: "Started refresh for ActiveEncode job #{status.active_encode_id}"
   end
 
+  # PATCH/PUT /admin/asset_files/ab2323ac/submit_hocr_and_textonly_pdf
+  def submit_hocr_and_textonly_pdf
+    @asset = Asset.find_by_friendlier_id!(params[:id])
+    authorize! :update, @asset
+    unless params[:hocr].present? &&  params[:textonly_pdf].present?
+      redirect_to admin_asset_url(@asset), flash: { error: "Please provide a textonly_pdf and an hocr." }
+      return
+    end
+
+    # validate hocr
+    hocr = params[:hocr].read
+    parsed_hocr = Nokogiri::XML(hocr) { |config| config.strict }
+    unless parsed_hocr.css(".ocr_page").length == 1
+      redirect_to admin_asset_url(@asset), flash: { error: "This HOCR file isn't valid." }
+      return
+    end
```

Comment:
I think you have to rescue a `Nokogiri::XML::SyntaxError` here, parallel to how you rescue PDF::Reader::MalformedPDFError below. 

Is this enough code that we should pull it out into a helper object? that then maybe can be refactored to be a bit cleaner. But doesn't need to be TOO much, not trying to add lots more work, just pulling out into a helper object would be good no matter what (and maybe adding like two tests of it? possibly replacing some of the controller tests?)